src/components/FlavorStash/Note.js

```diff
@@ -0,0 +1,183 @@
+    userId: PropTypes.string.isRequired
+  };
+
+  const dispatch = useDispatch();
+  const loading = useSelector(isLoading);
+  const note = useSelector(getFlavorNote);
+  const [editing, setEditing] = useState(false);
+
+  useEffect(() => {
+    dispatch(noteActions.requestNote({ flavorId }));
+  }, [dispatch]);
+
+  const handleToggleEditing = () => {
+    setEditing(!editing);
+  };
+
+  const addNoteIcon = (id) => {
+    return (
+      <Fragment>
+        {!editing && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => handleToggleEditing(id)}
+          >
+            <FontAwesomeIcon icon="plus" size="sm" title="Add Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  };
+
+  const editNoteIcon = (id) => {
+    return (
+      <Fragment>
+        {!editing && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => handleToggleEditing(id)}
+          >
+            <FontAwesomeIcon icon="pen" size="sm" title="Edit Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  };
+
+  const handleNoteSubmit = (values) => {
+    const { update } = values;
+    const noteEdit = values.note;
+
+    if (!noteEdit) {
+      dispatch(noteActions.deleteNote(values));
+    } else if (update === true) {
+      dispatch(noteActions.updateNote(values));
+    } else {
+      dispatch(noteActions.createNote(values));
+    }
+
+    handleToggleEditing();
+  };
+
+  const noteEditor = () => {
+    const update = Boolean(note?.[flavorId]?.note);
+
+    return (
+      <FinalForm
+        onSubmit={handleNoteSubmit}
+        initialValues={{
+          flavorId,
+          note: note?.[flavorId]?.note,
+          update,
+          userId
+        }}
+        render={({ handleSubmit, submitting, values }) => (
+          <Form noValidate onSubmit={handleSubmit}>
+            <Form.Row>
+              <Field name="note">
+                {({ input }) => (
+                  <Form.Group as={Col}>
+                    <InputGroup>
+                      <Form.Control
+                        {...input}
+                        as="textarea"
+                        placeholder="Flavor Note"
+                        rows="10"
+                      />
+                    </InputGroup>
+                  </Form.Group>
+                )}
+              </Field>
+            </Form.Row>
+            {values?.note ? (
+              <Form.Row>
+                <h4>Preview:</h4>
+                <Col md="12">
+                  {
+                    /* eslint-disable-next-line no-sync */
+                    unified()
+                      .use(parse)
+                      .use(remark2react)
+                      .processSync(values.note).result
+                  }
+                </Col>
+              </Form.Row>
+            ) : null}
+            <Field name="userId">

```

Comment:
Can these fields be pulled at submit-time rather than populated as hidden fields now?

Reply:
I removed them and added a commit, their values are pre-filled and submitted with the form. I mainly just had them there for readability in the generated page source.